File: server/src/models/education.rs
use serde::Serialize;
use sqlx::FromRow;

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct EducationModel {
    pub id: i32,
    pub user_id: i32,
    pub school: String,
    pub degree: Option<String>,
    pub field: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
    pub created_at: String,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct PublicEducationModel {
    pub school: String,
    pub degree: Option<String>,
    pub field: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
}

#[derive(Clone, FromRow, Debug, Serialize)]
pub struct AuthEducationModel {
    pub id: i32,
    pub school: String,
    pub degree: Option<String>,
    pub field: Option<String>,
    pub start_date: Option<String>,
    pub end_date: Option<String>,
}

File: src/types/education.ts
export interface EducationModel {
  id: number;
  user_id: number;
  school: string;
  degree?: string;
  field?: string;
  start_date?: string;
  end_date?: string;
  created_at: string;
}

export interface PublicEducationModel {
  school: string;
  degree?: string;
  field?: string;
  start_date?: string;
  end_date?: string;
}

export interface AuthEducationModel extends PublicEducationModel {
  id?: number;
}

export interface AddEducationPayload {
  school: string;
  degree?: string;
  field?: string;
  start_date?: string;
  end_date?: string;
}
